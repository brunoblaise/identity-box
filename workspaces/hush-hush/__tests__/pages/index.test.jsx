import { render, screen } from '@testing-library/react'

import Home from '../../pages/index'

describe('index', () => {
  beforeEach(() => {
    console.log = jest.fn()
    jest.useFakeTimers()
  })

  afterEach(() => {
    console.log.mockRestore()
  })

  it('has proper styling', async () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
    await screen.findByText(/^Connect\.\.\./i)
  })

  it('displays welcome text', async () => {
    render(<Home />)
    expect(screen.getByText(/Hush Hush needs to know two things/i)).toBeInTheDocument()
    await screen.findByText(/^Connect\.\.\./i)
  })
})
