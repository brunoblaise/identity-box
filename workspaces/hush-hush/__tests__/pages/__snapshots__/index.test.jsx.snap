// Vitest Snapshot v1

exports[`index > has proper styling 1`] = `
<div>
  <div
    style="display: flex; flex-flow: row nowrap; justify-content: center; align-items: center; height: 100vh; width: 100vw;"
  >
    <div
      style="display: flex; flex-flow: column nowrap; align-items: center; min-height: 150px; max-width: 550px; width: 85%;"
    >
      <div
        style="display: flex; flex-flow: column nowrap; align-items: flex-start; justify-content: center; width: 100%; opacity: 0;"
      >
        <div
          style="display: flex; width: 100%; flex-flow: column nowrap; align-items: center; justify-content: center;"
        >
          <div
            style="text-align: center; margin-top: 0; margin-bottom: 0; color: white;"
          >
            In order to share your secret securely with the intendent (and only intended) recipient, Hush Hush needs to know two things:
          </div>
          <div
            style="text-align: center; margin-top: 15px; margin-bottom: 0; color: white;"
          >
            1. The DID of the intended recipient.
          </div>
          <div
            style="text-align: center; margin-top: 15px; margin-bottom: 0; color: white;"
          >
            2. Your current DID.
          </div>
          <div
            style="text-align: center; margin-top: 20px; margin-bottom: 0; color: white;"
          >
            For this reason, Hush Hush, needs to connect to your mobile where you can then select the intended recipient from your 
            <span
              style="color: #0099FF;"
            >
              Identity Box IdApp Address Book
            </span>
            .
          </div>
          <div
            style="height: 20px; width: 1px;"
          />
          <div
            style="text-align: center; margin-top: 0; margin-bottom: 0; color: white;"
          >
            Please make sure that you have your 
            <span
              style="color: #0099FF;"
            >
              IdApp
            </span>
             ready and that you have selected the identity you want to use before hitting 
            <span
              style="color: #0099FF;"
            >
              connect
            </span>
             below.
          </div>
          <div
            style="height: 50px; width: 1px;"
          />
          <div
            style="align-self: center;"
          >
            <button
              class="ui primary button"
            >
              Connect...
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
